webrender/src/hit_test.rs

```diff
@@ -505,8 +505,12 @@ impl HitTester {
         result
     }
 
-    pub fn get_pipeline_root(&self, pipeline_id: PipelineId) -> &HitTestSpatialNode {
-        &self.spatial_nodes[self.pipeline_root_nodes[&pipeline_id].0 as usize]
+    pub fn get_pipeline_root(&self, pipeline_id: PipelineId) -> Option<&HitTestSpatialNode> {
```

Comment:
what are the conditions for which the pipeline root couldn't be found? I'm worried that by returning an `Option` we'd just be hiding an internal logic problem.

Reply:
Unfortunately, I'm not sure what are the conditions under which this can generally happen as I'm not that familiar to the codebase.

Reply:
More importantly, I wonder why Servo hits this and Gecko doesn't. Perhaps, it's something on Servo side to fix... We need to understand the problem better.

Reply:
Yes, I'm digging in a bit in https://github.com/servo/servo/issues/23964 since it's easy to reproduce the panic.

Reply:
Based on my investigations in that issue, I think silently suppressing this error is probably not the best solution.